ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -901,26 +901,28 @@ private Offset generateReferenceIndexSegmentStartOffset(Offset recordOffset) {
   List<IndexEntry> getValidIndexEntriesForIndexSegment(Offset indexSegmentStartOffset, long deleteReferenceTimeMs,
       long expiryReferenceTimeMs) {
     List<IndexEntry> validEntries = new ArrayList<>();
-    for (Map.Entry<MockId, IndexValue> indexSegmentEntry : referenceIndex.get(indexSegmentStartOffset).entrySet()) {
-      MockId key = indexSegmentEntry.getKey();
-      IndexValue value = indexSegmentEntry.getValue();
-      if (value.isFlagSet(IndexValue.Flags.Delete_Index)) {
-        // delete record is always valid
-        validEntries.add(new IndexEntry(key, value));
-        if (!isDeletedAt(key, deleteReferenceTimeMs) && !isExpiredAt(key, expiryReferenceTimeMs)
-            && value.getOriginalMessageOffset() != IndexValue.UNKNOWN_ORIGINAL_MESSAGE_OFFSET
-            && value.getOriginalMessageOffset() >= indexSegmentStartOffset.getOffset()) {
-          // delete is irrelevant but it's in the same index segment as the put and the put is still valid
-          validEntries.add(new IndexEntry(key, allKeys.get(key).getFirst()));
-        }
-      } else if (!isExpiredAt(key, expiryReferenceTimeMs)) {
-        // unexpired
-        if (!deletedKeys.contains(key)) {
-          // non expired, non deleted PUT
-          validEntries.add(new IndexEntry(key, value));
-        } else if (!isDeletedAt(key, deleteReferenceTimeMs)) {
-          // delete does not count
+    if (referenceIndex.containsKey(indexSegmentStartOffset)) {
```

Comment:
why would this not be true? The else on this seems like it should throw an exception.